subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/TestEventSerializer.java

```diff
@@ -183,12 +182,19 @@ public DefaultTestFailure read(Decoder decoder) throws Exception {
             for (int i = 0; i < numOfCauses; i++) {
                 causes.add(read(decoder));
             }
+
+            String rawFailureName = decoder.readString();
```

Comment:
🧐 Perhaps worth extracting a `readFailure()` method here?

Would also allow a good place to document the purpose of this check, which might seem to be just wrapping an exception for no good reason.

Reply:
💭 Alternately, might this new method be moved to the `Serializer` interface itself as a `default readCatching()` method?  It seems like we might run into this problem elsewhere.

Reply:
I've pulled it out into a method with some docs on why we do what we do.  

I'm not sure about pulling it up into `Serializer` because I couldn't think of a case where we would want to do something similar (i.e. catch the exception and wrap it rather than just throwing it up the chain).  Doesn't mean a case doesn't exist, I just couldn't think of one.